apps/managedidentity/managedidentity.go

```diff
@@ -50,8 +51,19 @@ const (
 	imdsAPIVersion = "2018-02-01"
 
 	systemAssignedManagedIdentity = "system_assigned_managed_identity"
+	defaultRetryCount             = 3
 )
 
+// retry on these codes
+var retryStatusCodes = []int{
```

Comment:
Slightly mismatched with .NET implementation: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/785fc1843e6244470db00237f2d0e020c1180e95/src/client/Microsoft.Identity.Client/Http/HttpRetryCondition.cs#L25

Reply:
Updated the codes to match other MSAL's 

Azure sdk for go are from where the rest of the codes are 
https://github.com/Azure/azure-sdk-for-go/blob/3ebd0d439f9d8aa06a0764a60892a8001b997a34/sdk/azidentity/managed_identity_client.go#L114